### PR TITLE
Add ability to select node filter using config options

### PR DESCRIPTION
**Pull request by AlexeyKasatkin**
Closes: #77

### CODE REVIEW

File: cmd/app/options.go [hunk 1]
```diff
@@ -45,6 +47,7 @@ func (o *options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.Mask, "mask", "32", "mask part of the cidr")
 	fs.StringVar(&o.Kubeconfig, "kubeconfig", "", "kubeconfig to use with kubernetes client")
 	fs.StringVar(&o.Hostname, "hostname", "", "We will use os.Hostname if none provided")
+	fs.StringVar(&o.NodeFilter, "nodefilter", "fair", "We will use 'fair' if none provided")
```
Comment (dshulyak) [inline cmd/app/options.go]: is there any way to declare options?

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -60,6 +61,15 @@ func NewIPClaimScheduler(config *rest.Config, mask string, monitorInterval time.
 		},
 	}
 
+	switch nodeFilter{
+	case "fair":
+		getNode := getFairNode
```
Comment (dshulyak) [inline pkg/scheduler/scheduler.go]: s.getFairNode ?

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -95,6 +107,8 @@ type ipClaimScheduler struct {
 	claimStore   cache.Store
 	serviceStore cache.Store
 
+	getNode([]*extensions.IpNode) (*extensions.IpNode)
```
Comment (dshulyak) [inline pkg/scheduler/scheduler.go]: nit: no need for parenthesis

**Comment by dshulyak** [comment]
Please modify one of e2e tests to use first node filter. Also I suggest to change name from first to first-live or first-alive, to make it clear that we won't choose node that is considered unreachable from ipcontroller point of view,

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -74,10 +75,24 @@ func NewIPClaimScheduler(config *rest.Config, mask string, monitorInterval time.
 		observedGeneration: make(map[string]int64),
 		liveIpNodes:        make(map[string]struct{}),
 
-		queue: workqueue.NewQueue(),
-	}, nil
+		//queue: workqueue.NewQueue(),
```
Comment (dshulyak) [inline pkg/scheduler/scheduler.go]: why?

File: pkg/scheduler/scheduler.go [hunk 5]
```diff
@@ -74,10 +75,24 @@ func NewIPClaimScheduler(config *rest.Config, mask string, monitorInterval time.
...
+		scheduler.getNode = scheduler.getFirstNode
+	default:
+		return nil, errors.New("Incorrect node filter is provided")
+	}
+
+	scheduler.queue = workqueue.NewQueue()
+	return &scheduler, nil
 }
 
+type NodeFilter func([]*extensions.IpNode) *extensions.IpNode
```
Comment (dshulyak) [inline pkg/scheduler/scheduler.go]: should be private

File: cmd/app/options.go (hunk 1 above)
Comment (AlexeyKasatkin) [inline cmd/app/options.go]: We can list them in help string and check manually. I've found examples where it is done. So, I suppose, it is not done automatically.

**Pull request merged**